### ISSUE TITLE
Windows PortableApps.com PAF format portable version

### ISSUE THREAD

**Issue by stephenjudge** [canton7/SyncTrayzor#152]
Would there be any chance the portable version could be packaged in the PortableApps.com PAF format and published to their directory so it can be downloaded, installed and updated easily by us PortableApps users?

http://portableapps.com/development/portableapps.com_format

(Their website seems to be having a moment, so I'd look at that page in a few hours or days).

**Comment by canton7** [comment]
It should be do-able (the paths to use e.g. for configuration are set in `SyncTrayzor.exe.config`, so they can be tweaked without recompiling), but I can't say I've ever created a PAF format application before. I'll take a closer look when I get some free time (or feel free to get something working, and I can integrate it into the build system :wink:)

**Comment by canton7** [comment]
Right, there's very little motivation on my end to do this: I'm not a portable apps user.

If you can put together an example of what needs to happen - what needs to be packed where, and what the paths need to be changed to in `SyncTrayzor.exe.config`, I'm more than happy to automate it and make it part of the build process.

**Comment by nigelhooper** [comment]
Just registering my interest in this issue.  Unfortunately I have no experience of the creating a PAF format application and it doesn't look like a 5 minute job, if I get some spare time I will see if I can work out an example.

**Comment by marclaporte** [comment]
PortableApps is awesome even if you don't need portability on a USB key.

It reduces management overhead of a Windows instance.
* All your apps installed and upgraded with a single process
* It makes software just as easy to manage as files
** If you re-image your Windows or change machines, you just move your PortableApps directory (or restore from backup), and voilà, you are ready to go.
** Deleting an app is as simple as deleting the directory and there is no residue (ex. phantom dlls)
* Looking for an app? The apps are vetted by a community and if one turns nasty, it's excluded.

Please see benefits: http://portableapps.com/about/what_is_a_portable_app

This would bring more users to Syncthing.

In the PortableApps ecosystem, there are 900k downloads per week (yes, nearly 1 million per week!)
http://sourceforge.net/projects/portableapps/

I wonder if Syncthing would do a good job of keeping PortableApps software in sync? I suspect this would depend on each software. In any case, it's easy to let each instance upgrade the apps.

Best regards,

M ;-)

**Comment by canton7** [comment]
The advantages sound similar to chocolatey then. 

As I said above, i'm definitely open to having a PortableApps package, but there's no personal motivation: I don't have enough free time to read up on the format when there's no personal gain. If someone else looks into it and makes the necessary tweaks to the config (and figures out what to do with regards to e.g. the auto-upgrade mechanism) I'll be more than happy to integrate it.

**Comment by marclaporte** [comment]
I am interested in funding this development. Please contact me directly if you have the skills and interest but need some funding.

Thanks!

(Potential co-funders are welcome as well to contact me)

**Comment by 2wayne** [comment]
I've already made a Syncthing Portable: http://portableapps.com/node/54229
Unfortunately SyncTrayzor depends on .NET which isn't portable. Imho the browser is the best portable gui ;)

**Comment by canton7** [comment]
Fair logic. I'd say most PCs will have at least 4.5.2 on them, but whether or not that's there's a rule against portable apps that may or may not run successfully depending on the setup of the target machine I don't know.

**Comment by antonio-gil** [comment]
Nice work @2wayne ! I'm looking forward to test your portable version.

By the way, @2wayne and @canton7 , what could be the difference between the portable and the standalone versions of the application?

**Comment by canton7** [comment]
This issue is solely about having a version which works with [PortableApps](https://portableapps.com/): they have their own application format, package managers, etc, as far as I understand.

The standalone version of SyncTrayzor will work just fine from a USB stick if you don't use PortableApps.
